libs/community/langchain_community/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,288 @@
+                'process' methods are used. Defaults to 1.
+            show_progress: If true, will show a progress bar as the files are
+                loaded. This forces an iteration through all matching files to
+                count them prior to loading them.
+
+        Raises:
+            ValueError: if the length of the lists in loader_args are not equal
+        """
+        loader_args = list(zip(*init_loader_args.values()))
+        self.loaders_args = [
+            dict(zip(init_loader_args.keys(), loader_arg)) for loader_arg in loader_args
+        ]
+        self.loader_callable = loader_initializer_callable
+        self.show_progress = show_progress
+        self._partial_load = partial(_load_loader, loader_callable=self.loader_callable)
+        self.method = method
+        self.max_workers = max_workers
+
+        # Check that all lists in loader_args have the same length
+        if len(set(len(arg) for arg in init_loader_args.values())) != 1:
+            raise ValueError("All loader_args lists must have the same length")
+
+    def load(self) -> List[Document]:
+        """Load data from the loader using the specified method. The list of
+        documents is flattened.
+
+        Possible methods are:
+            - "sequential": load data sequentially
+            - "thread": load data using threads
+            - "process": load data using processes
+            - "async": load data using asyncio (use the `aload` method instead)
+
+        Args:
+            method (str, optional): concurrent method to use.
+                Defaults to "sequential".
+            max_workers (int, optional): number of workers if 'thread' or
+                'process' methods are used. Defaults to 1.
+
+        Raises:
+            ValueError: if the method is not valid
+
+        Returns:
+            List[Document]: list of loaded documents
+        """
+
+        if self.method == "thread":
+            return self._load_thread(self.max_workers)
+        elif self.method == "process":
+            return self._load_process(self.max_workers)
+        elif self.method == "sequential":
+            return self._load_sequential()
+        elif self.method == "async":
+            raise ValueError(
+                "You must use the `aload` method to use the 'async' method"
+            )
+        else:
+            raise ValueError(f"Invalid method {self.method}")
+
+    async def aload(self) -> List[Document]:
+        """Load data from the loader using asyncio.
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+        return await self._async_load()
+
+    async def _async_load(self) -> List[Document]:
+        """Load data from the loader using asyncio.
+
+        Returns:
+            List[Document]: list of loaded documents flattened.
+        """
+
+        event_loop = asyncio.get_event_loop()
+        tasks = [
+            event_loop.run_in_executor(
```

Comment:
This defeats the goal of asyncio of not using threads.
The `alazy_load` method of the underlying loaders should be called instead.